lib/ddtrace/writer.rb

```diff
@@ -117,10 +117,16 @@ def stats
 
     private
 
-    def sampling_updater(response)
+    def sampling_updater(response, api)
       return unless response.is_a?(Net::HTTPOK)
-      service_rates = JSON.parse(response.body)
-      @priority_sampler.update(service_rates)
+
+      if api[:version] == HTTPTransport::V4
```

Comment:
that should do the trick, but I agree with you that this coupling between `Writer` and `Transport` makes the code brittle and should be addressed in a future PR.